File: LayoutTests/http/tests/security/resources/subresource2/protected-image.php
<?php
header("Cache-Control: no-store");
header("Connection: close");
if (!isset($_SERVER["PHP_AUTH_USER"])) {
    header("WWW-authenticate: Basic realm=\"" . $_SERVER["REQUEST_URI"] . "\"");
    header("HTTP/1.0 401 Unauthorized");
    exit;
}
// Authenticated
header("Content-Type: image/png");
echo file_get_contents("../../contentSecurityPolicy/block-all-mixed-content/resources/red-square.png");
?>

File: LayoutTests/http/tests/security/resources/subresource1/protected-image.py
#!/usr/bin/env python3

import base64
import os
import sys

username = base64.b64decode(os.environ.get('HTTP_AUTHORIZATION', ' Og==').split(' ')[1]).decode().split(':')[0]

uri = os.environ.get('REQUEST_URI', '')

sys.stdout.write(
    'Cache-Control: no-store\r\n'
    'Connection: close\r\n'
)

if not username:
    sys.stdout.write(
        'WWW-authenticate: Basic realm="{}"\r\n'
        'Content-Type: text/html\r\n'
        'status: 401\r\n\r\n'.format(uri)
    )
    sys.exit(0)

sys.stdout.write('Content-Type: image/png\r\n\r\n')
sys.stdout.flush()
with open(os.path.join(os.path.dirname(__file__), '../../contentSecurityPolicy/block-all-mixed-content/resources/red-square.png'), 'rb') as file:
    sys.stdout.buffer.write(file.read())
